File: languages/java/src/main/java/com/bloock/sdk/entity/identity_v2/BooleanAttributeDescriptor.java
package com.bloock.sdk.entity.identity_v2;

import com.bloock.sdk.bridge.proto.IdentityEntitiesV2;
import com.bloock.sdk.bridge.proto.IdentityEntitiesV2.BooleanAttributeDefinitionV2;

public class BooleanAttributeDescriptor extends AttributeDescriptor {
  public BooleanAttributeDescriptor(
      String displayName, String technicalName, String description, Boolean required) {
    super(displayName, technicalName, description, required);
  }

  public static BooleanAttributeDescriptor fromProto(BooleanAttributeDefinitionV2 res) {
    return new BooleanAttributeDescriptor(
        res.getDisplayName(), res.getId(), res.getDescription(), res.getRequired());
  }

  public BooleanAttributeDefinitionV2 toProto() {
    return IdentityEntitiesV2.BooleanAttributeDefinitionV2.newBuilder()
        .setDisplayName(this.displayName)
        .setId(this.technicalName)
        .setDescription(this.description)
        .setRequired(this.required)
        .build();
  }
}

File: languages/php/Bloock/Entity/IdentityV2/BooleanAttributeDescriptor.php
<?php

namespace Bloock\Entity\IdentityV2;

use Bloock\BooleanAttributeDefinitionV2;

class BooleanAttributeDescriptor extends AttributeDescriptor
{
    public static function fromProto(BooleanAttributeDefinitionV2 $res): BooleanAttributeDescriptor
    {
        return new BooleanAttributeDescriptor($res->getDisplayName(), $res->getId(), $res->getDescription(), $res->getRequired());
    }

    public function toProto(): BooleanAttributeDefinitionV2
    {
        $p = new BooleanAttributeDefinitionV2();
        $p->setDisplayName($this->displayName);
        $p->setId($this->technicalName);
        $p->setDescription($this->description);
        $p->setRequired($this->required);
        return $p;
    }
}
